### PR TITLE
Add testPropertyIO

### PR DESCRIPTION
**Pull request by tranma**
This is useful when you want to use custom generators:

```
prop_foo
  = testPropertyIO
  $ do x <- pick myCustomGeneratorThatIsNotAnArbitraryInstance
       ...
```

Also eliminates the need to use `monadicIO` because I keep making the mistake of using `monadicIO $ return ...` which makes the test always passes.

### CODE REVIEW

File: disorder-core/src/Disorder/Core.hs [hunk 1]
```diff
@@ -7,3 +7,4 @@ import           Disorder.Core.OrdPair as X
 import           Disorder.Core.Property as X
 import           Disorder.Core.Tripping as X
 import           Disorder.Core.UniquePair as X
+import           Disorder.Core.IO as X
```
Comment (charleso) [inline disorder-core/src/Disorder/Core.hs]: I think not having this in intentional, but not sure

File: disorder-core/src/Disorder/Core/IO.hs [hunk 2]
```diff
@@ -13,6 +14,9 @@ import           System.CPUTime (getCPUTime)
 testIO :: Testable a => IO a -> Property
 testIO = monadicIO . (=<<) stop . run
 
+testPropertyIO :: Testable a => PropertyM IO a -> Property
+testPropertyIO = monadicIO . (=<<) stop
```
Comment (charleso) [inline disorder-core/src/Disorder/Core/IO.hs]: Worth calling from `testIO`?

**Comment by charleso** [comment]
Apart from the `Core.hs` change :+1:

**Comment by tranma** [comment]
I want to add a test like this but it's a bit silly

```Haskell
prop_falseFails :: Property
prop_falseFails
  = expectFailure $ testPropertyIO (return False)

prop_falseDoesNotFail :: Property
prop_falseDoesNotFail
  = monadicIO $ liftIO $ return False
```

File: disorder-core/src/Disorder/Core.hs (hunk 1 above)
Comment (tranma) [inline disorder-core/src/Disorder/Core.hs]: oh ok, will remove

File: disorder-core/test/Test/Disorder/Core/IO.hs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import           Test.QuickCheck.Monadic
+
+prop_falseFails :: Property
+prop_falseFails
+  = expectFailure $ testPropertyIO $ return False
+
+prop_falseDoesNotFail :: Property
+prop_falseDoesNotFail
+  = monadicIO $ liftIO $ return False
+
```
Comment (tranma) [inline disorder-core/test/Test/Disorder/Core/IO.hs]: these might already be obvious /notsure

**Comment by tranma** [comment]
merging this

I kinda want a warning whenever "monadicIO" is used.

**Pull request merged**